api/test/e2enew/route_online_debug/route_online_debug_test.go

```diff
@@ -0,0 +1,785 @@
+			Headers: map[string]string{
+				"Authorization":                 base.GetToken(),
+				"online_debug_url":              base.APISIXInternalUrl + `/hello`,
+				"online_debug_request_protocol": "http",
+				"online_debug_method":           "GET",
+				"Content-Type":                  "multipart/form-data",
+				"online_debug_header_params":    `{"test":["test1"],"Authorization": ["Basic amFjazoxMjM0NTYKIA=="]}`,
+			},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   `"data":{"code":200,"message":"200 OK","data":"hello world`,
+		}),
+		table.Entry("online debug without basic-auth", base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPost,
+			Path:   "/apisix/admin/debug-request-forwarding",
+			Headers: map[string]string{
+				"Authorization":                 base.GetToken(),
+				"online_debug_url":              base.APISIXInternalUrl + `/hello`,
+				"online_debug_request_protocol": "http",
+				"online_debug_method":           "GET",
+				"Content-Type":                  "multipart/form-data",
+				"online_debug_header_params":    `{"test":["test1"]}`,
+			},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   `{"code":401,"message":"401 Unauthorized","data":{"message":"Missing authorization in request"}}`,
+		}),
+		table.Entry("delete the route just created", base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/routes/r1",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		}),
+		table.Entry("hit the route just deleted", base.HttpTestCase{
+			Object:       base.APISIXExpect(),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			ExpectStatus: http.StatusNotFound,
+			ExpectBody:   "{\"error_msg\":\"404 Route Not Found\"}\n",
+			Sleep:        base.SleepTime,
+		}),
+		table.Entry("delete consumer", base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/consumers/jack",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		}),
+	)
+
+	table.DescribeTable("Route_Online_Debug_Route_With_Key_Auth",
+		func(tc base.HttpTestCase) {
+			base.RunTestCase(tc)
+		},
+
+		table.Entry("hit route that not exist", base.HttpTestCase{
+			Object:       base.APISIXExpect(),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			ExpectStatus: http.StatusNotFound,
+			ExpectBody:   "{\"error_msg\":\"404 Route Not Found\"}\n",
+		}),
+		table.Entry("create route enable key-auth plugin", base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPut,
+			Path:   "/apisix/admin/routes/r1",
+			Body: `{
+				"uri": "/hello",
+				"methods": ["GET"],
+				"plugins": {
+					"key-auth": {}
+				},
+				"upstream": {
+					"type": "roundrobin",
+					"nodes": [{
+						"host": "172.16.238.20",
```

Comment:
replace this IP in this file too, @Jaycean, thanks.

Reply:
done. Thks.